### ISSUE TITLE
Including options with injectManifest

### ISSUE THREAD

**Issue by Nooshu** [GoogleChrome/workbox#1453]
**Library Affected**:
WorkBox 3.1.0

**Browser & Platform**:
All browsers

**Issue or Feature Request Description**:
Hopefully this is a simple yes no answer. I've searched all over the web but haven't been able to find an example of this. Is it possible to include options with the `workbox.precaching.precacheAndRoute([])`?

For example if I wanted to include the `cleanUrls` option, how would I do that? With injectManifest this fails as I guess it isn't exactly what it is looking for.
```js
workbox.precaching.precacheAndRoute([], {/* additional options here */})
````

Is the only option here to use the `getManifest` to generate the array then pass it into `precacheAndRoute()` along with my custom options? And if so does anyone have an example of this setup?

Thanks!

**Comment by jeffposnick** [comment]
As of the `v3.1.0` release, the RegExp used by default to find the "injection point" is https://github.com/GoogleChrome/workbox/blob/77b64a7b956aad7c1627aa5e4b3c3d8f48c36c43/packages/workbox-build/src/entry-points/options/defaults.js#L27

That's a little hard to parse by eye, but it includes a requirement that there's a trailing `)` after the `[]` placeholder array. That prevents you from including your own options as you're attempting to do.

Thinking back to the origins of this pattern, I'm pretty sure that we came up with it back when there wasn't an optional second parameter could be passed in to `precacheAndRoute()`. We added in that optional second parameter later on, but didn't adjust the default RegExp to make it more lenient.

You can explicitly override the default RegExp by configuring `injectionPointRegexp` along with your other build options:

```
{
  injectionPointRegexp: new RegExp('(\.precacheAndRoute\()\s*\[\s*\]\s*(\)|,)'),
  // ... other Workbox options....
}
```

(That differs from the default RegExp in that it will match *either* a `)` *or* a `,` following the `[]`.)

Can you confirm that that modified RegExp gives you the behavior you expect? If so, I think it can safely become the new default in a future release.

**Comment by Nooshu** [comment]
Hi Jeff,  Thanks for getting back to me.

So I just tried your modified `injectionPointRegexp`. The build now passes with no error. Unfortunately the resulting service worker is now no longer valid as there is an additional array argument and a missing comma between the array and the resulting additional options:

**Input**
````javascript
workbox.precaching.precacheAndRoute([], { cleanUrls: true })
````

**Output**
````javascript
workbox.precaching.precacheAndRoute([],[
// generated objects here...
] { cleanUrls: true })
````

**Comment by jeffposnick** [comment]
Ooops, there was some escaping of the `\` that got lost in translation when I moved from using `/ ... /` literal syntax to `new RegExp(string)` syntax.

You can try with

```js
injectionPointRegexp: /(\.precacheAndRoute\()\s*\[\s*\]\s*(\)|,)/
```

I just tested that out and it seems to work. It would be great if you could confirm as well.

Stay tuned for a PR to make that change to the default.

**Comment by Nooshu** [comment]
Hi @jeffposnick, Just to confirm I am now getting the expected output

**Config**
````javascript
injectionPointRegexp: /(\.precacheAndRoute\()\s*\[\s*\]\s*(\)|,)/
````

**Input**
````javascript
workbox.precaching.precacheAndRoute([], { cleanUrls: true })
````

**Output**
````javascript
workbox.precaching.precacheAndRoute([
// generated objects here...
], { cleanUrls: true })
````

Thanks for the quick feedback! Looking forward to seeing it by default in the next version.
